### DISCUSSION TITLE
Upgrade used JSS Angular Version to 16

### DISCUSSION THREAD

**Discussion by m7alshehri** [Sitecore/jss#1499] [Ideas]
### Is your suggestion related to a problem? Please describe.

soon used version 14 will not get any more security updates, because the security support is end of life is at (2023-11-18).
Also, frameworks that may be dependent on the angular version, will no longer get updates for older versions.
It's also not possible to use newer framework features with the old version.

### Describe the solution you'd like

I would like to request an update of the used angular version to the newest version 16.x.

### Is there a way to achieve this functionality with exisiting tools?

no alternative

### Additional information

Is there a roadmap for that?

**Comment by art-alexeyenko** [comment]
Angular 14 is still marked as LTS, but we do need to upgrade it.
v15 update is in our pipeline for this year (certainly before the end of LTS for v14), and perhaps we'll jump right to v16.

**Reply by tomvoigt** [reply]
@art-alexeyenko are you also aware that NodeJS 16.x will be EOL on 11-09-2023? According to https://angular.io/guide/versions we can start running NodeJS 18.x after we have upgraded to Angular 15. This is currently not possible, since `sitecore-jss-angular` does not yet support Angular 15.

Skipping Angular 15 would severely complicate our LCM, so this is not preferred on our end.

**Reply by art-alexeyenko** [reply]
@tomvoigt  The decision to move node 16 EOL was a bit of a blindside, can't lie. We aim to roll out Angular 15 support soon, in the next release, so please be on the lookout for that.

**Reply by informatious** [reply]
@art-alexeyenko why did you choose to go for Angular 15 and not immediately for 16? Features in 15 seem limited compared to those in 16 and LTS for 15 already ends 2024-05-18.
Is there any target date for the next Sitecore release?
Thanks in advance for your reply.

**Reply by art-alexeyenko** [reply]
@informatious The release of v15 support has just been rolled out, and we'd like to also roll out the support for v16 by the end of the year (hopefully earlier, but I'm just being careful with timelines).
The reason we went for v15 immediately is to address the expedited nodejs 16 EOL and have all our samples be compatible with node 18. Specifically because v15 changes are more limited, we used it as a sort of stepping maintenance stone to be set up for v16 update later.

**Reply by tomvoigt** [reply]
@art-alexeyenko Great! Looking forward to that, much appreciated.

**Reply by kpicavet** [reply]
@art-alexeyenko We did the upgrade to angular 15, but when trying to build the application we get an error on the build:server script saying it doesn't know the argument 'bundle-dependencies' . 

I tried to do the same in the sample app (seems like it's still on 14, but upgraded it as well) & got the same result:

`ng run sitecore-jss-app:server:production --output-path=$npm_package_config_buildArtifactsPath/server --output-hashing=none --bundle-dependencies true`

=> Error: Unknown argument: bundle-dependencies

If I remove the argument & deploy it, our app is broken. Any idea how to solve this?

**Reply by AVierwind** [reply]
Is the end of the year still the planning for 16?

**Reply by art-alexeyenko** [reply]
@AVierwind the work on the update is planned to be done within the next two weeks - and it should go into 21.6 release in January.
v21.6 will have some extra XMCloud-centric things that we want to ensure are working correctly, in addition to the Angular update. I'm sorry about the delay on this, but I ask you to please bear a little longer. It would help us ensure a more smooth release overall.

**Comment by kpicavet** [comment]
Angular 16 introduces a lot of nice new features like signals and improvements for server side rendering.
Especially the server side rendering improvements would give the performance of an angular app a huge boost which we would need to make our case to start building our app with angular + jss.

**Comment by AVierwind** [comment]
The difference between 15 and 16 in functionality is quite big. It is not be possible to have a angular 16 project using this package until this package is also on v16. This is currently stalling our LCM. 

Also i am curious what the strategy is for this type of life cycle management on the angular package, so we can plan our lcm projects accordingly

**Comment by illiakovalenko** [comment]
Hey guys
Update regarding this discussion: we upgraded our sdk and template to **Angular 16**.
At this moment it's available within the **canary** version

**Comment by spike1292** [comment]
Created a pr to fix some regression issues we found during upgrading: https://github.com/Sitecore/jss/pull/1706
